engine/src/main/java/org/hibernate/search/util/impl/FileHelper.java

```diff
@@ -84,58 +84,50 @@ else if ( !destination.isDirectory() ) {
 		}
 	}
 
-	public static void synchronize(File source, File destination, boolean smart) throws IOException {
+	public static void synchronize(Path source, Path destination, boolean smart) throws IOException {
 		synchronize( source, destination, smart, DEFAULT_COPY_BUFFER_SIZE );
 	}
 
-	public static void synchronize(File source, File destination, boolean smart, long chunkSize) throws IOException {
+	public static void synchronize(Path source, Path destination, boolean smart, long chunkSize) throws IOException {
 		if ( chunkSize <= 0 ) {
 			log.checkSizeMustBePositive();
 			chunkSize = DEFAULT_COPY_BUFFER_SIZE;
 		}
-		if ( source.isDirectory() ) {
-			if ( !destination.exists() ) {
-				if ( !destination.mkdirs() ) {
-					throw new IOException( "Could not create path " + destination );
-				}
-			}
-			else if ( !destination.isDirectory() ) {
-				throw new IOException(
-						"Source and Destination not of the same type:"
-								+ source.getCanonicalPath() + " , " + destination.getCanonicalPath()
-				);
-			}
-			String[] sources = source.list();
-			Set<String> srcNames = new HashSet<String>( Arrays.asList( sources ) );
-			String[] dests = destination.list();
+		if ( Files.isDirectory( source ) ) {
+			Files.createDirectories( destination );
+			Set<Path> sources = Files.list( source ).collect( Collectors.toSet() );
+			Set<String> sourceFilenames = sources.stream().map( v -> v.getFileName().toString() ).collect( Collectors.toSet() );
+			Set<Path> dests = Files.list( destination ).collect( Collectors.toSet() );
+			Set<String> destFilenames = dests.stream().map( v -> v.getFileName().toString() ).collect( Collectors.toSet() );
 
 			//delete files not present in source
-			for ( String fileName : dests ) {
-				if ( !srcNames.contains( fileName ) ) {
-					delete( new File( destination, fileName ) );
+			for ( String fileName : destFilenames ) {
+				if ( !sourceFilenames.contains( fileName ) ) {
+					delete( destination.resolve( fileName ) );
 				}
 			}
 			//copy each file from source
-			for ( String fileName : sources ) {
-				File srcFile = new File( source, fileName );
-				File destFile = new File( destination, fileName );
+			for ( Path srcFile : sources ) {
+				Path destFile = destination.resolve( srcFile.getFileName() );
 				synchronize( srcFile, destFile, smart, chunkSize );
 			}
 		}
 		else {
-			if ( destination.exists() && destination.isDirectory() ) {
-				tryDelete( destination.toPath() );
-			}
-			if ( destination.exists() ) {
-				long sts = source.lastModified() / FAT_PRECISION;
-				long dts = destination.lastModified() / FAT_PRECISION;
+			if ( Files.exists( destination ) && Files.isDirectory( destination ) ) {
+				tryDelete( destination );
+			}
+			final File sourceFile = source.toFile();
+			final File destinationFile = destination.toFile();
```

Comment:
ok, but I just wanted to remove a `File` reference from a deprecated SPI and look what it got me into :)